lib/api/apiUtils/object/partInfo.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Checks for partNumber in request query and returns the part number
+ * @param {object} query - request query
+ * @return {(Integer|undefined)} - part number, zero if part number is less than
+ * 1 and undefined if no partNumber is found in the request query
+ */
+function getPartNumber(query) {
+    if (query && query.partNumber !== undefined) {
+        return Number.isNaN(query.partNumber) ?
+            0 : Number.parseInt(query.partNumber, 10);
+    }
+    return undefined;
+}
+
+/**
+ * Gets the size of the requested part of the object
+ * @param {object} objMD - object metadata
+ * @param {object} partNumber - part number
+ * @return {(Integer|undefined)} - size of the part or undefined
+ */
+function getPartSize(objMD, partNumber) {
```

Comment:
This is actually a way more efficient and less confusing way of doing it, but just for future reference, we do store part info in metadata as StoredParts, returned with this function: https://github.com/scality/cloudserver/blob/development/8.2/lib/services.js#L650